pkg/cache/clusterqueue.go

```diff
@@ -62,8 +63,11 @@ type ClusterQueue struct {
 	NamespaceSelector labels.Selector
 	Preemption        kueue.ClusterQueuePreemption
 	FlavorFungibility kueue.FlavorFungibility
-	AdmissionChecks   sets.Set[string]
-	Status            metrics.ClusterQueueStatus
+	// Aggregates AdmissionChecks from both .spec.AdmissionChecks and .spec.AdmissionCheckStrategy
+	// Sets hold ResourceFlavors to which an AdmissionCheck should apply.
+	// In case its empty, it means an AdmissionCheck should apply to all ResourceFlavor
+	AdmissionChecks map[string]sets.Set[string]
```

Comment:
```suggestion
	AdmissionChecks map[string]sets.Set[kueue.ResourceFlavorReference]
```

I'm wondering if we can avoid casting at some points.
@PBundyra WDYT?

Reply:
Good idea, thanks!